src/scanner.cc

```diff
@@ -339,6 +346,11 @@ struct Scanner {
       advance(lexer);
     }
 
+    // Handles :!~
+    if (starts_with_bang && lexer->lookahead == '~') {
+      advance(lexer);
+    }
```

Comment:
Should this logic be part of `scan_operator`? It seems like that method deals with most of the operator symbols like `:+`.

Reply:
Good call. Updated.